controllers/image_generator.go

```diff
@@ -133,6 +134,12 @@ func ImageCreate(c client.Client) (int, error) {
 	}
 
 	ig := GetImageGenerator()
```

Comment:
What about extending GetImageGenerator to error when ig is nil? Then manage the error here. This could make it clearer what to expect for future interactions with this code.

Reply:
Having a re-thought. Ideally if the InitializeImageGenerator -> newImageGenerator has succeeded then ig shouldn't be nil. Which means GetImageGenerator should always return a non nil ig.
I believe the core issue is the InitializeImageGenerator not being run subsequently in case of error from newImageGenerator
How about changing the method to something like this
```
func InitializeImageGenerator(client client.Client) error {
	var err error
	igOnce.Do(func() {
		ig, err = newImageGenerator(client)
	})
        if err != nil { 
            igOnce = sync.Once{} 
        }
	return err
}
```

Reply:
Thanks Pradipta and Camilla, will build a new image with these changes and try it out on libvirt env..

Reply:
So basically @ajayvic you'll need to make changes to InitializeImageGenerator as suggested above.
Also you'll need to change the newImageGenerator as I had mentioned earlier by removing the following line in the default case handling
```
		return nil, fmt.Errorf("unsupported cloud provider: %s", ig.provider)
```

Reply:
Thanks Pradipta. Refactored the code as per the suggestions..
Kindly review..